src/validation/EmailValidator.java

```diff
@@ -0,0 +1,71 @@
+package validation;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class EmailValidator {
+
+	private String allowedSpecialSymbols = "~!@#$%^&*_+=-`'./{}";
+	private List<String> tldList;
+	
+	public boolean hasAtSign(String string) {
+		if (string.indexOf("@") == -1) return false;
+		return true;
+	}
+
+	public boolean hasInvalidCharacters(String string) {
+	    if (string.contains("..")) return true;
+		
+		
+		for(int i=0; i < string.length(); i++) {
+			char c = string.charAt(i);
+			if(!Character.isLetterOrDigit(c) && allowedSpecialSymbols.indexOf(c) == -1) return true;
+		}
+		
+		String[] split = string.split("@");
+		int lenght = split[0].length();
+		if (split[0].charAt(0) == '.' && split[0].charAt(lenght - 1) == '.') return true;
+		
+		return false;
+	}
+
+	public boolean hasCorrectDomainAndTDL(String string) {
+		if (!hasAtSign(string)) return false;

```

Comment:
That check should not be part of this function. It can be independent and called by validateEmail public function.